descope/internal/mgmt/fga.go

```diff
@@ -0,0 +1,93 @@
+package mgmt
+
+import (
+	"context"
+
+	"github.com/descope/go-sdk/descope"
+	"github.com/descope/go-sdk/descope/api"
+	"github.com/descope/go-sdk/descope/internal/utils"
+	"github.com/descope/go-sdk/descope/sdk"
+)
+
+type fga struct {
+	managementBase
+}
+
+var _ sdk.FGA = &fga{}
+
+func (f *fga) SaveSchema(ctx context.Context, schema *descope.FGASchema) error {
+	if schema == nil {
+		return utils.NewInvalidArgumentError("schema")
+	}
+	body := map[string]any{
+		"dsl": schema.Schema,
+	}
+	_, err := f.client.DoPostRequest(ctx, api.Routes.ManagementFGASaveSchema(), body, nil, f.conf.ManagementKey)
+	return err
+}
+
+func (f *fga) CreateRelations(ctx context.Context, relations []*descope.FGARelation) error {
+	if len(relations) == 0 {
+		return utils.NewInvalidArgumentError("relations")
+	}
+
+	body := map[string]any{
+		"tuples": relations,
+	}
+
+	_, err := f.client.DoPostRequest(ctx, api.Routes.ManagementFGACreateRelations(), body, nil, f.conf.ManagementKey)
+	return err
+}
+
+func (f *fga) DeleteRelations(ctx context.Context, relations []*descope.FGARelation) error {
+	if len(relations) == 0 {
+		return utils.NewInvalidArgumentError("relations")
+	}
+
+	body := map[string]any{
+		"tuples": relations,
+	}
+
+	_, err := f.client.DoPostRequest(ctx, api.Routes.ManagementFGADeleteRelations(), body, nil, f.conf.ManagementKey)
```

Comment:
delete use post?

Reply:
This is delete of multiple items, not like RESTful delete of a single item in the URL, we do it in many places